lib/pbench/server/database/models/__init__.py

```diff
@@ -55,23 +57,31 @@ def process_result_value(self, value, dialect):
 
 
 def decode_integrity_error(
-    exception: IntegrityError, on_null: Callable, on_duplicate: Callable
+    model: Database.Base,
+    exception: Exception,
```

Comment:
The point was that we try to "decode" whatever we get back in most cases; it might be some other `SQLAlchemyError`. So I loosened the signature to make tools happy and confirmed the actual type before acting on it.

Reply:
OK.  Should probably tweak the argument description in the docstring, then.